Add error handling for unsupported hash method

fluteze opened the pull request:
When using MySQL database for passwords, an unsupported or typo'ed hash method name was causing user passwords to be deleted with only the contents of the $PREFIX to be stored. i.e. password field in the database would have '{ARGON2ID}' but no following text to contain the password.

fluteze committed:
Added error handling for unsupported hash methods.

Handle the case where the password hash is not found or is unsupported. Generate an error message.
Note: When using the MySQL driver (and possibly others) with an unsupported hash or a typo in the hash name, the user's password was being erased in the database.

fluteze committed:
Update en_US.inc

johndoh commented on plugins/password/localization/en_US.inc:
```diff
@@ -25,6 +25,7 @@ $messages['nopassword'] = 'Please input new password.';
 $messages['nocurpassword'] = 'Please input current password.';
 $messages['passwordincorrect'] = 'Current password incorrect.';
 $messages['passwordinconsistency'] = 'Passwords do not match, please try again.';
+$messages['passwdhashnotsupported'] = 'Password hash function unknown or unsupported. Possible typo in hash name.');
```
There is an extra ')' on this line

fluteze replied:
Oops. Removed the extra )

fluteze replied:
This is my first time submitting a PR update via github. Do I need to do something to have the update propagate?

fluteze committed:
Update en_US.inc

Removed extraneous )

alecpl commented on plugins/password/password.php:
```diff
@@ -769,6 +773,14 @@ static function hash_password($password, $method = '', $prefixed = true)
+            rcube::raise_error(array(
+                'code' => 600, 'file' => __FILE__, 'line' => __LINE__,
+                'message' => "Password plugin: Hash method not supported."
+                ), true, true);
```
The way you call raise_error() make it stop script execution. And this is good. We don't want to display the error to the end user. Please, remove use of PASSWORD_HASH_NOT_SUPPORTED and the following return statement.

fluteze replied:
Done.

fluteze committed:
Removed error message to user

The script will stop on misconfiguration w/o notifying the user.

fluteze committed:
Removed unused message

Message removed as the administrator log is the only one to notify rather than the user.
